.github/workflows/ci.yaml

```diff
@@ -218,6 +218,39 @@ jobs:
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
 
+  mutation-test:
+    name: Mutation Testing
+
+    if: github.event_name != 'schedule'
+
+    needs:
+      - end-to-end-tests
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install PHP with extensions
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.3
+          coverage: pcov
+          extensions: none, ctype, curl, dom, json, libxml, mbstring, phar, soap, tokenizer, xml, xmlwriter
+          ini-values: assert.exception=1, zend.assertions=1, error_reporting=-1, log_errors_max_len=0, display_errors=On
+          tools: none
+
+      - name: Install infection
```

Comment:
I understand that `roave-infection-static-analysis-plugin` can only be installed using Composer. However, installing Infection and the plugin using Composer for every run feels wrong. May I suggest that we create a `tools/infection` directory, put a `composer.json` file there that pulls in what we need, and then put the entire directory (`composer.json`, `composer.lock`, `vendor`) under version control? Then, just like with all other development tools, everything we need is available after a clone/checkout.

Reply:
I think I can live with this approach, at least for now.

Reply:
That being said, we need to install Infection and the plugin into an empty directory with a separate `composer.json` / `composer.lock` combo. Otherwise the actual `composer.json` / `composer.lock` used in this repository will interfere, as can be seen here https://github.com/sebastianbergmann/phpunit/actions/runs/8511186285/job/23310422850.

I know it's not helpful, but I truly hate installing development tools using Composer :-)

Reply:
In all fairness, Infection to this day still autoload the project's code as it relies on reflection. Could be solved by leveraging BetterReflection or publishing a scoped PHAR like PHPStan does, but that's still work to do.

But if you can live with that risk then using a separate composer.json is still fine.